source/common/filesystem/subscription_impl.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "envoy/config/subscription.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/filesystem/filesystem.h"
+
+#include "common/common/macros.h"
+#include "common/config/utility.h"
+#include "common/filesystem/filesystem_impl.h"
+
+#include "api/base.pb.h"
+#include "google/protobuf/util/json_util.h"
+
+namespace Envoy {
+namespace Filesystem {
+
+/**
+ * Filesystem inotify implementation of the API Subscription interface. This allows the API to be
+ * consumed on filesystem changes to files containing the JSON canonical representation of
+ * lists of ResourceType.
+ */
+template <class ResourceType> class SubscriptionImpl : public Config::Subscription<ResourceType> {
```

Comment:
Sorry maybe I wasn't paying attention, but it feels like all config impls should be inside the config namespace so they would be inside `/source/common/config/...` It feels like this would colocate a lot of related config loading code even if the transports are the same, whereas the other namespaces are for more low level stuff. 

This is really just how I logically view things. Don't feel super strongly if you feel otherwise.